sdk/cosmos/azure-cosmos/azure/cosmos/_cosmos_client_connection.py

```diff
@@ -155,7 +157,7 @@ def __init__(
         }
 
         # Keeps the latest response headers from the server.
-        self.last_response_headers: Dict[str, Any] = {}
+        self.last_response_headers: CaseInsensitiveDict = CaseInsensitiveDict()
```

Comment:
Is the reason we've droped the subtyping because CaseInsensitiveDict is already typed as `[str, Any]`?

Reply:
I think it was in part that and in part pylint - can try re-adding it if we'd think it's a good idea to have it again